providers/tentacle.rb

```diff
@@ -129,6 +129,83 @@
   new_resource.updated_by_last_action(actions_updated?([install, create_home_dir, generate_cert, create_instance, configure, service]))
 end
 
+action :register do
+  new_resource = @new_resource
+  instance = new_resource.instance
+  port = new_resource.port
+  polling_port = new_resource.polling_port
+  polling = fancy_bool(new_resource.polling)
+  server = new_resource.server
+  api_key = new_resource.api_key
+  roles = new_resource.roles
+  environment = new_resource.environment
+
+  verify_server(server)
+  verify_api_key(api_key)
+  verify_roles(roles)
+  verify_environment(environment)
+
+  firewall = windows_firewall_rule 'Octopus Deploy Tentacle' do
+    localport port.to_s
+    dir :in
+    protocol 'TCP'
+    firewall_action :allow
+    action :create
+    only_if { polling == 'False' }
+  end
+
+  # # Itterate over every role and make a big long string like:
+  # # --role "web" --role "database" --role "app-server"
+  role_expression = ''
+  roles.each do |a_role|
+    role_expression << "--role \"#{a_role}\" "
+  end
+
+  register_listening_instance = powershell_script "register-listening-tentacle-instance-#{instance}" do
```

Comment:
I'm going to defer the abstraction of the --comms style since its adding complexity and I can't quite get it working right.